crates/sargon/src/system/sargon_os/transactions/sargon_os_transaction_analysis.rs

```diff
@@ -0,0 +1,705 @@
+                radix_engine_toolkit_receipt: None,
+                logs: vec![],
+                receipt: TransactionReceipt {
+                    status: TransactionReceiptStatus::Succeeded,
+                    error_message: None,
+                },
+            },
+        );
+        let os =
+            prepare_os(MockNetworkingDriver::new_with_bodies(200, responses))
+                .await;
+        let manifest = TransactionManifest::set_owner_keys_hashes(
+            &IdentityAddress::sample().into(),
+            AccountOrPersona::sample_mainnet_other()
+                .virtual_hierarchical_deterministic_factor_instances()
+                .into_iter()
+                .map(|i| PublicKeyHash::hash(i.public_key.public_key))
+                .collect(),
+        );
+
+        let result = os
+            .perform_transaction_preview_analysis(
+                manifest.instructions_string(),
+                Blobs::sample(),
+                Message::sample(),
+                false,
+                Nonce::sample(),
+                PublicKey::sample(),
+            )
+            .await;
+
+        assert_eq!(
+            result,
+            Err(CommonError::FailedToExtractTransactionReceiptBytes)
+        )
+    }
+
+    #[actix_rt::test]
+    async fn execution_summary_parse_error() {
+        let responses = prepare_responses(
+            LedgerState {
+                network: "".to_string(),
+                state_version: 0,
+                proposer_round_timestamp: "".to_string(),
+                epoch: 0,
+                round: 0,
+            },
+            TransactionPreviewResponse {
+                encoded_receipt: "".to_string(),
+                radix_engine_toolkit_receipt: Some(
+                    ScryptoSerializableToolkitTransactionReceipt::Reject {
+                        reason: "Test".to_string(),
+                    },
+                ),
+                logs: vec![],
+                receipt: TransactionReceipt {
+                    status: TransactionReceiptStatus::Succeeded,
+                    error_message: None,
+                },
+            },
+        );
+        let os =
+            prepare_os(MockNetworkingDriver::new_with_bodies(200, responses))
+                .await;
+
+        let result = os
+            .perform_transaction_preview_analysis(
+                TransactionManifest::sample().instructions_string(),
+                Blobs::sample(),
+                Message::sample(),
+                false,
+                Nonce::sample(),
+                PublicKey::sample(),
+            )
+            .await;
+
+        assert_eq!(
+            result,
+            Err(CommonError::ExecutionSummaryFail {
+                underlying: "InvalidReceipt".to_string()
+            })
+        )
+    }
+
+    #[actix_rt::test]
+    async fn execution_summary_reserved_instructions_error() {
+        let responses = prepare_responses(
```

Comment:
many `ScryptoDecimal192::zero().into(),`, I think it derives `Copy` so you might wanna do a `let zero: DecimalType = ScryptoDecimal192::zero();` in the top here and use `zero` variable. 

You are calling into, which type are we changing into? A `RET` Decimal?

Reply:
Good point! Yeap, we're changing them into `RET` Decimal.

Reply:
`let ret_zero: AsStr<Decimal> = Decimal::ZERO.into();`
This is the best I could do, am I missing something? Please let me know if there is a better and simpler way to do this.